stdlib/LinearAlgebra/src/lu.jl

```diff
@@ -215,14 +233,18 @@ lupivottype(::Type{T}) where {T} = RowMaximum()
 
 # for all other types we must promote to a type which is stable under division
 """
-    lu(A, pivot = RowMaximum(); check = true) -> F::LU
+    lu(A, pivot = RowMaximum(); check = true, allowsingular = false) -> F::LU
 
 Compute the LU factorization of `A`.
 
 When `check = true`, an error is thrown if the decomposition fails.
 When `check = false`, responsibility for checking the decomposition's
 validity (via [`issuccess`](@ref)) lies with the user.
 
+By default with `check = true` an error is also thrown when the decomposition
```

Comment:
It'll be good to have an example demonstrating the use of `allowsingular = true`.

Reply:
There's the following a the bottom of the `lu` docstring:

```
julia> lu([1 2; 1 2], allowsingular = true)
LU{Float64, Matrix{Float64}, Vector{Int64}}
L factor:
2×2 Matrix{Float64}:
 1.0  0.0
 1.0  1.0
U factor (rank-deficient):
2×2 Matrix{Float64}:
 1.0  2.0
 0.0  0.0
```

Do you mean something else?

Reply:
Oh I had missed that. This looks fine.